core/src/main/java/com/flipkart/varadhi/core/cluster/ClusterManager.java

```diff
@@ -0,0 +1,24 @@
+package com.flipkart.varadhi.core.cluster;
+
+import java.util.List;
+
+/**
+ * This interface is responsible for managing the cluster membership. Its implementation is responsible for capturing the
+ * node resources, local ip & hostname, so that they can be used while registering to the cluster and other nodes can
+ * be notified of the same.
+ */
+public interface ClusterManager {
+    void join();
+
+    void leave();
+
+    List<NodeInfo> getAllMembers();
+
+    void addMembershipListener(MembershipListener listener);
+
+    NodeConnection connect(String nodeId);
```

Comment:
https://docs.google.com/document/d/1iyE0c38rWc-_t6iWwwp_i-MJKz1G4VyNSh1sELAFnlw/edit

This method is not present in the contract

What is the requirement of this connect method?

Reply:
This method was a attempt to create an api to create p2p connections to other node in the cluster. This is already being refined by Dhruv in the doc:
https://docs.google.com/document/d/1CPOPcM05TufnsXKoD0FrEqoXrH6xecVYk4NDkg4dFus/

So who ever is picking the implementation part of it, can refactor this part.